packages/datadog-instrumentations/src/fastify.js

```diff
@@ -27,6 +28,11 @@ function wrapFastify (fastify, hasParsingEvents) {
       app.addHook('preHandler', preValidation)
     }
 
+    // No need to add the onRoute hook unless Code Origin for Spans is enabled
+    if (codeOriginForSpansChannel.hasSubscribers) {
```

Comment:
This should be in the hook instead because subscribers can be enabled/disabled at runtime.

Reply:
The hook is only fired at boot when the routes are registered (not sure if there's a use-case for registering routes later though 🤔). So I put it here to not be too noisy and add hooks that were not needed. But since it's not a hot code-path (as it normally only happens at boot), I can move it 👍 

In what scenarios will this channel have its subscriber added or removed at runtime? I know that it technically can, but in this case we control when it happens, so shouldn't it be ok?

Reply:
Actually, scratch that. I think you idea of making this a generic "route has been added" channel is really great and it removes the coupling with the specific feature, which I wasn't happy about to begin with. I'll update the code 👍

Reply:
I've pushed a commit with a fix. Let me know if that's what you had in mind